src/lib.rs

```diff
@@ -203,23 +204,125 @@ impl From<std::io::ErrorKind> for Error {
     }
 }
 
+bitflags! {
+    /// Network device flags.
+    pub struct IfFlags: u32 {
```

Comment:
I would not name it `IfFlags` but `DeviveFlags` or just `Flags`.

Reply:
Just `Flags` is not clear enough as there's also the `DeviceFlags` struct. `ConnectionFlags` does not reflect what it represents which is the interface itself which is why I've chosen `IfFlags`. However, I'm still open to other suggestions. Perhaps renaming `DeviceFlags` to `Flags` and the `IfFlags` to `DeviceFlags`?